contracts/core/src/octane/Staking.sol

```diff
@@ -150,18 +150,13 @@ contract Staking is OwnableUpgradeable, EIP712Upgradeable {
 
     /**
      * @notice Increase your validators self delegation.
-     *         NOTE: Only self delegations to existing validators are currently supported.
-     *         If msg.sender is not a validator, the delegation will be lost.
      * @dev Proxies x/staking.MsgDelegate
      */
-    function delegate(address validator) external payable {
+    function delegate(address delegator, address validator) external payable {
```

Comment:
I'd suggest rather creating a new method, since the signature is changing. This means that bindings will not be compatbile with our onchain contracts, which isn't good. Main brnach code should remain backwards compatible with onchain deployed contracts. Once they have been upgraded, then only can we remove methods that we dont need.

Reply:
we should have called the method `selfDelegate`, then we could simply have added `delegate` now... 🤦

Reply:
we don't need to change the signature.

This should be keep `delegate(address validator)`, we just have to remove the check below that enforces self delegation.

We do not want anyone to be able to choose delegations for anyone else

Reply:
Rolled the function back to its original implementation, minus the self-delegation check. Implemented a new delegateFor function. Changes are reflected in the Go codebase as well, restoring those as they were before.